internal/controlplane/handlers_repositories.go

```diff
@@ -491,28 +455,54 @@ func (s *Server) ListRemoteRepositoriesFromProvider(
 	return out, nil
 }
 
-func (s *Server) deleteRepositoryAndWebhook(
+// TODO: this probably can probably be used elsewhere
+// returns project ID and github client - this flow of code exists in multiple
+// places
+func (s *Server) getProjectIDAndClient(
 	ctx context.Context,
-	repo db.Repository,
-	provider db.Provider,
-) error {
-	tx, err := s.store.BeginTransaction()
+	request HasProtoContext,
+) (uuid.UUID, v1.GitHub, error) {
+	projectID := getProjectID(ctx)
+
+	provider, err := getProviderFromRequestOrDefault(ctx, s.store, request, projectID)
+	if err != nil {
+		return uuid.Nil, nil, providerError(err)
+	}
+
+	pbOpts := []providers.ProviderBuilderOption{
+		providers.WithProviderMetrics(s.provMt),
+		providers.WithRestClientCache(s.restClientCache),
+	}
+
+	p, err := providers.GetProviderBuilder(ctx, provider, s.store, s.cryptoEngine, pbOpts...)
 	if err != nil {
-		return status.Errorf(codes.Internal, "error deleting repository")
+		return uuid.Nil, nil, status.Errorf(codes.Internal, "cannot get provider builder: %v", err)
 	}
-	defer s.store.Rollback(tx)
 
-	qtx := s.store.GetQuerierWithTransaction(tx)
-	if err := qtx.DeleteRepository(ctx, repo.ID); err != nil {
-		return status.Errorf(codes.Internal, "error deleting repository: %v", err)
+	client, err := p.GetGitHub()
+	if err != nil {
+		return uuid.Nil, nil, status.Errorf(codes.Internal, "error creating github provider: %v", err)
 	}
 
-	if err := s.deleteWebhookFromRepository(ctx, provider, repo); err != nil {
+	return projectID, client, nil
+}
+
+// covers the common logic for the two varieties of repo deletion
+func (s *Server) deleteRepository(
+	ctx context.Context,
+	request HasProtoContext,
+	deletionMethod func(v1.GitHub, uuid.UUID) error,
+) error {
+	projectID, client, err := s.getProjectIDAndClient(ctx, request)
+	if err != nil {
 		return err
 	}
 
-	if err := s.store.Commit(tx); err != nil {
-		return status.Errorf(codes.Internal, "error deleting repository")
+	err = deletionMethod(client, projectID)
```

Comment:
The indirection here seems less useful than the other parts of this PR, but I may be missing something.

It feels like the two methods should be:

```
DeleteRepoByName() {
  repo := getRepoByName()
  deleteRepo(repo.id)
}

DeleteRepoById() {
  deleteRepo(req.id)
}
```

Reply:
While we certainly can pass one function to another, it feels like it might be simpler to just pass data around and have one "standard" form.

Reply:
(if `deleteRepo` checks project ownership, the first form doesn't even need to check that the name lines up, though that may be the course of least resistance.)